scripts/gptGenerateJsonFromIssue.py

```diff
@@ -2,120 +2,116 @@
-            "facebook": "https://www.facebook.com/example",
-            "drive": "https://drive.google.com/drive/folders/example",
-            "form": "https://docs.google.com/forms/d/example/viewform",
-            "docs": "https://docs.google.com/forms/d/example/viewform",
-            "tags": ["Social", "Instagram", "Facebook", "Twitter", "TikTok", "Advocacy"]
-        }
+        openai.api_key = api_key or os.environ.get('OPENAI_API_KEY')
+        if openai.api_key is None:
+            raise ValueError("No API key provided. Set the OPENAI_API_KEY environment variable or pass the api_key argument.")
 
-        Using the provided record example structure, new initiative info to generate a new record by adhering to the
-        following guidelines:
+    def ask_gpt_for_initiative_json(self, issue_body: str) -> str:
+        query =  f"""
+        Please provide arguments for the add_new_initiative function using the following guidelines:
 
-        1. Remove Empty Keys: Ensure that no key in the record has an empty or null value.
-        2. Tag Optimization: add hebrew tags about the relevant initiative, the tags should aid in efficient searching.
-        3. If possible, provide the "displayName" and "description" fields in Hebrew.
-        4. Ensure that the "description" field is between 110-130 characters in length.
-        5. If exists add initiativeValidationDescription in hebrew upto between 110-130 chars
-        6. If a value for a key is empty or "No response" don't add that key to the output response.
-        7. Important, please limit the response only to the output json.
-        """
+        * Ensure that no parameter has an empty, null or "no response" value.
+        * Add hebrew tags about the relevant initiative, the tags should aid in efficient searching.
+        * If possible, provide the "displayName" and "description" fields in Hebrew.
 
-        formatted_query = base_query + \
-            '\n\n new initiative info is:\n```' + text_string + '```'
-        return formatted_query
+        The details of the new initiative to be added have been provided in the following form, provided in markdown syntax:
+        {issue_body}
+        """
+        logging.info("GPT query: " + query)
 
-    def query_via_api(self, text_string: str) -> str:
-        query = self.format_query(text_string)
-        conversation = [
-            {"role": "system", "content": "You are a helpful assistant."},
+        messages = [
+            {"role": "system", "content": "You are only used for function calling, and must exclusively use the function you have been provided with"},
             {"role": "user", "content": query}
         ]
-        response = openai.ChatCompletion.create(
-            model=self.model,
-            messages=conversation
-        )
-        answer = response['choices'][0]['message']['content']
-        return answer
-
-    def query_via_rest(self, text_string: str) -> str:
```

Comment:
I don't see why - under the covers the SDK also issues REST requests, except it potentially wraps them with retries, logging, possibly multiple requests for a single logical operation, etc.

TLDR the whole point of a good SDK is that it abstracts the underlying REST API for you (otherwise we wouldn't ever need SDKs)